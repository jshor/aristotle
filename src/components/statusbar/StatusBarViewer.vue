<template>
  <div class="status-bar-viewer">
    <div class="status-bar-viewer__left">
      <slot name="left" />
    </div>
    <div class="status-bar-viewer__right">
      <slot name="right" />
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'StatusBarViewer'
})
</script>

<style lang="scss">
.status-bar-viewer {
  display: flex;
  height: 2em;
  align-items: center;
  background-color: var(--color-bg-secondary);
  color: var(--color-secondary);
  box-shadow: 0 0 $border-width var(--color-shadow);
  box-sizing: border-box;
  font-size: 0.85em;

  &__left, &__right, &__label {
    display: flex;
    align-items: center;
    height: 100%;
  }

  &__left {
    padding: 0 0.5em;
    flex: 1;
  }

  &__right {
    justify-content: flex-end;
    flex: 1;
  }

  &__label {
    text-align: center;
  }
}
</style>
