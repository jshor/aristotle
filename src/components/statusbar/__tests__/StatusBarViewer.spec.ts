import { shallowMount, VueWrapper } from '@vue/test-utils'
import { ComponentPublicInstance } from 'vue'
import StatusBarViewer from '../StatusBarViewer.vue'

describe('Status Bar Viewer', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(StatusBarViewer)
  })

  it('should match the snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
