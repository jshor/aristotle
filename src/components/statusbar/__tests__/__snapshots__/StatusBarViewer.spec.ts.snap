// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Status Bar Viewer > should match the snapshot 1`] = `
"<div class="status-bar-viewer">
  <div class="status-bar-viewer__left"></div>
  <div class="status-bar-viewer__right"></div>
</div>"
`;
